### DISCUSSION TITLE
Slow system over time

### DISCUSSION THREAD

**Discussion by Svardsten53** [mcguirepr89/BirdNET-Pi#1165] [General]
I have two installations that have been running for about two years and a friend also has an installation that has been running for a long time and we are increasingly starting to notice delays in certain functions. My installations have 450000 - 950000 detections in the database and the size of it is 100 -200 MB. All installations use SSD.

What starts to go slowly is the following and it may not be the same in all systems, but one or more of these errors occur.

- The response time before the log is displayed is about 60 seconds, but then everything is displayed normally.

- When many detections occur at the same time, the analysis function cannot catch up, a backlog is created and sometimes it hangs completely. Restarting Analysis Server helps for a while, but the error returns.
 
- When you click on Services in the Tools menu, you may have to wait several minutes before it appears.

It seems that there is a delay in the system for some functions, otherwise it works fine. The question is whether the error is in the BirdNET-Pi or in the operating system. All upgrades are installed for both.

Has anyone else experienced the same errors or any idea what could be causing the problems? Could some log file or similar have become too large and needs to be cleared?

I would very much appreciate comments on this. Thank you!

**Comment by Nachtzuster** [comment]
I guess running `top` (via ssh) would be a good first step to figure out what is eating cpu cycles (if anything).
Maybe some screenshots of top while you're running these slow operations?

Also I'm wondering about extra stuff you might have like Apprise, Flickr, BirdWeather, is any of that enabled?

**Reply by Svardsten53** [reply]
By using the `top` command I found that the log is created by `journalctl`. Command `journalctl --disk-usage` showed that the log files together used 4 GB! I reduced the size by `journalctl --vacuum-size=200M` and the problems with the long log response time and Services were solved.

Remaing is the issue with backlog in the analysis, but we have to wait until we have a large number of birds chirping before we can see if the reduced log size also solves this.

Anyway, it sounds promising.

**Reply by Nachtzuster** [reply]
I'm curious: did reducing the journal solve all the issues you saw permanently or did you have to take additional steps?

**Reply by Svardsten53** [reply]
The journalctl command solved the issues with slow log and hangup when choosing services in the menu. I have not been able to test backlog of analysis on my own system yet. A lot of work was done on the other system with the backlog issue to find the cause, but all ened up in reinstall of BirdNET-Pi and now it seems to work.
